app/mailers/notify_mailer.rb

```diff
@@ -112,9 +112,10 @@ def export_email(user, attachment)
     mail(to: @user.email, subject: "The export of your content is ready")
   end
 
-  def tag_moderator_confirmation_email(user, tag)
+  def tag_moderator_confirmation_email(user, tag, channel_slug)
     @tag = tag
     @user = user
+    @channel_slug = channel_slug
```

Comment:
I see that `@channel_slug is set to be used in the email templates, but how is it passed in? Ruby/Rails newb question probably. 😉

Reply:
You'll see in assign_tag_moderator in the file above, the channel slug gets passed in from NotifyMailer.tag_moderator_confirmation_email(user, tag, chat_channel_slug(tag)).deliver. 

Then, when we set channel_slug to an instance variable i.e. `@channel_slug`. Rails is able to access instance variables in the view.

Hope that answers your question?

Reply:
@nickytonline Rails has a concept called "view contexts". When calling `render`, this gets instantiated like this (code from `ActionView::Rendering`):

```ruby
    def view_context
      view_context_class.new(lookup_context, view_assigns, self)
    end
```

`view_assingns` is a helper defined in `AbstractController::Rendering`):

```ruby
    def view_assigns
      variables = instance_variables - _protected_ivars

      variables.each_with_object({}) do |name, hash|
        hash[name.slice(1, name.length)] = instance_variable_get(name)
      end
    end
```

So the instance variables essentially get copied from the controller to the view context, which is why you can access them in the views.